File: receiver-windows/src/OpenAudioLink/Protocol/AudioPayloadValidator.cs
namespace OpenAudioLink.Protocol
{
    public static class AudioPayloadValidator
    {
        public static void ValidateAacPayload(byte[] payload)
        {
            if (payload == null)
            {
                throw new PacketParseException("Payload is null.");
            }

            if (payload.Length < ProtocolConstants.AudioPayloadHeaderSize)
            {
                throw new PacketParseException("Payload is shorter than AAC header.");
            }

            if (payload[0] != ProtocolConstants.CodecAacLc)
            {
                throw new PacketParseException("Unsupported codec.");
            }

            uint encodedSize = PacketParser.ReadUInt32(payload, 15);
            if ((ulong)payload.Length != (ulong)ProtocolConstants.AudioPayloadHeaderSize + encodedSize)
            {
                throw new PacketParseException("AAC payload length mismatch.");
            }
        }
    }
}

File: sender-android/app/src/main/java/com/openaudiolink/protocol/AudioPayloadValidator.kt
package com.openaudiolink.protocol

object AudioPayloadValidator {
    fun validateAacPayload(payload: ByteArray?) {
        if (payload == null) throw PacketParseException("Payload is null.")
        if (payload.size < ProtocolConstants.AudioPayloadHeaderSize) {
            throw PacketParseException("Payload is shorter than AAC header.")
        }
        if ((payload[0].toInt() and 0xff) != ProtocolConstants.CodecAacLc) {
            throw PacketParseException("Unsupported codec.")
        }

        val encodedSize = PacketParser.readUInt32(payload, 15)
        if (payload.size.toLong() != ProtocolConstants.AudioPayloadHeaderSize.toLong() + encodedSize) {
            throw PacketParseException("AAC payload length mismatch.")
        }
    }
}
